### PR TITLE
Complete Chapter 1

### PR DESCRIPTION
**Pull request by danielboros**
Sorry for the dup PR. I think I accidentally opened it against the main repo. Oops. 😨 

### Solutions for Chapter 1

Thanks for the effort in putting this together! I'm liking it so far.

I'm running on Windows on my work laptop so I just copied and pasted the `cabal` test commands from the `Makefile` directly into my Powershell as a stop gap from not having `make` and they all passed. Then I realized I could install `make` with Chocolatey (same as the official way of installing Haskell on Windows) and that worked OK for running the various targets. Just FYI since there weren't any Windows instructions.

I had to do [this](https://www.stackage.org/blog/2020/08/ghc-8-10-2-windows-workaround) to deal with a GHC 8.10.2 issue on Windows which prevented the test suite from building correctly. Probably should have just used Linux...

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -472,8 +473,9 @@ Implement a function that returns the last digit of a given number.
   results. Or you can try to guess the function name, search for it and check
   whether it works for you!
 -}
--- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit n = mod x 10
+  where x = abs n
```
Comment (vrom911) [inline src/Chapter1.hs:478]: Nice usage of `where`! 👍🏼

File: src/Chapter1.hs [hunk 2]
```diff
@@ -536,8 +538,13 @@ value after "=" where the condition is true.
...
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | ((x > y) && (x < z) || (x > z) && (x < y))  = x
+    | ((y > x) && (y < z) || (y > z) && (y < x))  = y
+    | ((x == y) || (x == z))                      = x 
+    | ((y == z))                                  = y
```
Comment (vrom911) [inline src/Chapter1.hs]: Great solution!

Small side note: no need to specify parenthesis around the top expression. You can just say, for example:

```haskell
    | y == z = y
```

File: src/Chapter1.hs [hunk 3]
```diff
@@ -615,8 +625,12 @@ Try to introduce variables in this task (either with let-in or where) to avoid
...
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
-
+sumLast2 :: Int -> Int
+sumLast2 n = 
+    let absN            = abs n
+        lastDgt         = mod absN 10
+        secondToLastDgt = mod (div absN 10) 10
+        in (lastDgt + secondToLastDgt)
```
Comment (vrom911) [inline src/Chapter1.hs:633]: Nice one 👌🏼 

A few additional notes, no action needed, just FYI 🙂 
For `lastDgt` you could reuse the `lastDigit` function implemented above. As well as in the `secondToLastDgt`. You can see that both of them are using `mod ... 10`.

File: src/Chapter1.hs [hunk 4]
```diff
@@ -635,8 +649,12 @@ Implement a function that returns the first digit of a given number.
...
 aren't ready for this boss yet!
 -}
-
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n
+    | x < 10 = x
+    | otherwise = firstDigit (div x 10)
+  where
+    x = abs n
```
Comment (vrom911) [inline src/Chapter1.hs:657]: Elegant solution 😍

**Review by vrom911 — APPROVED** [review]
Congrats with successful finishing of Chapter 1 🥳 
Amazing job!
And you nailed the advanced task 👏🏼 

Thanks a lot for sharing the experience on Windows! That is very valuable. We would add some of the notes of yours in README, so others on Windows would be aware of the issues! (And it should be totally fine to make this course on Windows, at least we would try to make it work! No need to grab Linux for that 😄 )

**Comment by vrom911** [comment]
Hey @danielboros !
I think that this PR is ready to merge 🚀 Well done!

**Comment by chshersh** [comment]
⚠️ By the way, per new rules of the Hacktoberfest event, you would need to add the `hacktoberfest` topic to this fork repo:

* https://hacktoberfest.digitalocean.com/hacktoberfest-update

All previous contributions are legit, but from now on it should have the label to participate.

File: src/Chapter1.hs (hunk 2 above)
Comment (danielboros) [reply src/Chapter1.hs]: Fixed. I was a little paren-happy.

File: src/Chapter1.hs (hunk 3 above)
Comment (danielboros) [reply src/Chapter1.hs:633]: Done.

**Comment by danielboros** [comment]
> 
> 
> ⚠️ By the way, per new rules of the Hacktoberfest event, you would need to add the `hacktoberfest` topic to this fork repo:
> 
>     * https://hacktoberfest.digitalocean.com/hacktoberfest-update
> 
> 
> All previous contributions are legit, but from now on it should have the label to participate.

Thanks! Done.

**Pull request merged**